tools/create_class_definitions.py

```diff
@@ -224,6 +232,47 @@ def export_code(self):
         return code
 
 
+class CompressedZSTD(construct.Tunnel):
+    def __init__(self, subcon, level: int = 3):
+        super().__init__(subcon)
+        import zstd
+        self.lib = zstd
+        self.level = level
+
+    def _decode(self, data, context, path):
+        return self.lib.decompress(data)
+
+    def _encode(self, data, context, path):
+        return self.lib.compress(data, self.level)
```

Comment:
is this not duplicated code?

Reply:
oh, yeah could move to construct_extensions

Reply:
Better yet, unused duplicated code :)